gui/src/components/Panel.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import IconButton from './IconButton.svelte';
+  import BackSVG from './mono/leftarrow.svelte';
+  import * as router from 'svelte-spa-router';
+
+  export let title: string | undefined = undefined;
```

Comment:
There is a shorthand for this:

```typescript
  export let title: string;
```

However, in this case, I think you want:

```typescript
export let title: string | null = null;
```

The reason being that you generally want to avoid using `undefined` as a value. In this case in particular, the goal is to convey "the title is set to none", not "the title has not yet been set".